src/components/tables/TableReducers.ts

```diff
@@ -101,7 +112,8 @@ export const tableReducer = (
         datePickerRangeId: getTableChildComponentId(action.payload.id, TableChildComponent.DATEPICKER_RANGE),
       };
     case TableActions.modifyState:
-      return action.payload.tableStateModifier(state);
+      selectedIds = state.data && state.data.selectedIds ? state.data.selectedIds : [];
```

Comment:
`updateSelectedIDs(action.payload.tableStateModifier(state), selectedIds)` would do, no need for extend